support/modules-artifact-gen/single-file-artifact-gen

```diff
@@ -178,47 +200,20 @@ else
   stat -c %a "${file}" > $permissions_file
 fi
 
-# Check the the passthrough_args and potentially modify them
-# to avoid conflicts or to let them override the already args
-# provided to mender-artifact
-# Runs in a subshell to allow overriding some parameters passed
-# to mender-artifact
-passthrough_args_modified=" "
-echo $passthrough_args | xargs -n 2 printf "%s %s\n" | (while read -r flag arg; do
-  if [ -n "$flag" ]; then
-    case $flag in
-      -T | --type)
-        echo "Error: Conflicting flag '$flag'. Already specified by the script."
-        exit 1
-        ;;
-      -o | --output-path)
-        output_path=$arg
-        ;;
-      -n | --name)
-        artifact_name=$arg
-        ;;
-      *)
-        passthrough_args_modified="$passthrough_args_modified $flag $arg"
-        ;;
-    esac
-  fi
-done
-
 mender-artifact write module-image \
   -T single-file \
-  $device_types \
+  "${device_types[@]}" \
   -o "$output_path" \
   -n "$artifact_name" \
   -f "$dest_dir_file" \
   -f "$filename_file" \
   -f "$permissions_file" \
   -f "$file" \
-  $passthrough_args_modified
+  "${passthrough_args[@]}"
 
 
 echo "Artifact $output_path generated successfully:"
 mender-artifact read $output_path
-# End of subshell
-)
+
 rm -rf $tmpdir
```

Comment:
looking at the diff it is really hard to understand where is the other opening `(` matching that one. and github forbid be to add this comment to the line above, saying the index of the line is wrong.